Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -108,16 +104,55 @@ - (NSURLSessionTaskState)state
 
 #pragma mark Helpers
 
+- (void)makeRequest
+{
+    __block CDTHTTPInterceptorContext *ctx =
+        [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    // We make sure all objects support `interceptRequestInContext:` during init.
+    for (NSObject<CDTHTTPInterceptor> *obj in self.requestInterceptors) {
+        ctx = [obj interceptRequestInContext:ctx];
+    }
+
+    __weak CDTURLSessionTask *weakSelf = self;
+    self.inProgressTask = [self.session
+        dataTaskWithRequest:ctx.request
+          completionHandler:^void(NSData *data, NSURLResponse *response, NSError *error) {
+            __strong CDTURLSessionTask *strongSelf = weakSelf;
+            if (strongSelf) {
+                ctx.response = response;
```

Comment:
Is there a good reason to keep the context around -- does the response interceptor need access to the request? Otherwise we're just holding onto memory needlessly (e.g., large POST body?)